packages/ckeditor5-coremedia-general-richtext-support/README.md

```diff
@@ -0,0 +1,244 @@
+know, is that any configuration for GRS (and thus GHS) applies to the Data View.
+
+Thus, when a RichText paragraph may have an additional attribute like `dir` it
+is insufficient just configuring the allowed attribute for element `<p>` in
+GHS: You must also register the same attribute for all elements in Data View,
+which will be mapped to `<p>` in Data later on. Thus, we will have to configure
+GHS similar to this:
+
+```javascript
+{
+  htmlSupport: {
+    allow: [{
+      name: /^(p|h[1-6])$/,
+      attributes: {
+        dir: /(rtl|ltr)/,
+      },
+    }],
+  },
+}
+```
+
+This is, what is done by the `GeneralRichTextSupport` plugin for all elements
+pre-configured in RichText Data Processing. But this does not know about any
+customizations such as introducing new elements.
+
+**What to do for custom elements and attributes?** Luckily, the GHS
+configuration is extensible. Thus, a plugin may enrich the configuration, and
+you may as well decide adding _late_ configuration, which is, when creating
+the `ClassicEditor` instance.
+
+Having this, introducing support for a new element such as `<mark>` from
+[Highlight feature] requires two steps, if you want to ensure, that attributes
+added via Unified API for example are not removed as soon as edited in CKEditor:
+
+1. Add data-processing to represent `<mark>` in CoreMedia RichText 1.0.
+2. Extend GHS configuration provided by GRS by the new element.
+
+If we decide representing `<mark>` as `<span class="mark">` you will require
+two configuration entries `coremedia:richtext` and `htmlSupport`:
+
+```javascript
+import CoreMediaRichText
+  from "@coremedia/ckeditor5-coremedia-richtext/CoreMediaRichText";
+import GeneralHtmlSupport
+  from "@ckeditor/ckeditor5-html-support/src/generalhtmlsupport";
+import GeneralRichTextSupport
+  from "@coremedia/ckeditor5-coremedia-general-richtext-support/GeneralRichTextSupport";
+import {replaceByElementAndClassBackAndForth}
+  from "@coremedia/ckeditor5-coremedia-richtext/rules/ReplaceBy";
+import {CoreMediaRichText10Dtd}
+  from "@coremedia/ckeditor5-coremedia-general-richtext-support/GeneralRichTextSupport";
+
+/* ... */
+
+ClassicEditor.create(document.querySelector('.editor'), {
+  plugins: [
+    CoreMediaRichText,
+    GeneralHtmlSupport,
+    GeneralRichTextSupport,
+    // ...
+  ],
+  "coremedia:richtext": {
+    rules: {
+      elements: {
+        mark: replaceByElementAndClassBackAndForth("mark", "span", "mark"),
+      },
+    },
+  },
+  htmlSupport: {
+    allow: [
+      {
+        name: "mark",
+        ...CoreMediaRichText10Dtd.attrs,
+      }
+    ],
+  },
+});
+```
+
+As you see, this already uses two convenience APIs provided by the CoreMedia
+plugins:
+
+* **`replaceByElementAndClassBackAndForth`:**
+
+    This adds a rule, which maps `<mark>` to `<span class="mark">` in
+    data-processing.
+
+* **`CoreMediaRichText10Dtd.attrs`:**
+
+    These are the attributes, which are marked as _allowed_ for `<span>`,
```

Comment:
No, it is perfectly fine. But it may miss some clarification: You map to `<span>`, and thus, you have to allow the same attributes for `<mark>` as for `<span>` (you inherit the attributes from `<span>` to `<mark>`.

I will take this as todo for further refactoring of GRS.